### ISSUE TITLE
Validation of Addresses data -- reference to cadastral parcel

### ISSUE THREAD

**Issue by MichalMed** [inspire-eu-validation/ets-repository#107]
http://inspire-sandbox.jrc.ec.europa.eu/etf-webapp/v2/TestRuns/EIDefaa367d-3779-4106-841d-642abb704b9b.html

XML document '550671.xml', Address 'AD.852929', property 'parcel': The property references a resource at 'http://services.cuzk.cz/wfs/inspire-cp-wfs.asp?service=WFS&VERSION=2.0.0&request=GetFeature&storedQuery_id=urn:ogc:def:query:OGC-WFS::GetFeatureById&Id=CP.2829868306', but the resource is not an 'CadastralParcel' or one of its known subtypes. Found: element 'member' in namespace 'http://www.opengis.net/wfs/2.0'.

But it really is CadastralParcel. 

We also have similar errors in cases of AdministrativeUnit, where it shall be address component, but we are referencing AdministrativeUnit feature from the theme Administrative Units. Same for Building/AbstractConstruction. It also appears within AdminUnitName and AddressComponent.

I feel that referencing is not proper, but in the end it is much better and more useful to use references to other INSPIRE features form other topics, that are referencable. Other option is to remake service to provide adminUnitName and AddressComponent via ad-hoc query, with link in its adminUnit property. But then it take more edges to get there going through feature with no additional information. What is clearer solution in your oppinion?

**Comment by MichalMed** [comment]
So the proper implementation of GetFeatureId shall have e.g. CadastralParcel as root element? If so, we're going to make it happen.

To you post scriptum: that is exactly the point. Our service for addresses provides addresses and addresses only as featureTypes. Is that wrong?

**Comment by cportele** [comment]
> So the proper implementation of GetFeatureId shall have e.g. CadastralParcel as root element? If so, we're going to make it happen.

Yes, see the [spec](http://docs.opengeospatial.org/is/09-025r2/09-025r2.html#192).

> To you post scriptum: that is exactly the point. Our service for addresses provides addresses and addresses only as featureTypes. Is that wrong?

An address must have one of more components (like AdminUnitName, PostalDescriptor), which are features, too. That is, if the WFS provides Address features it must also provide the component features. I think it should be fine to only list the Address feature type in the WFS capabilities, but the component features must be referenced either with an absolute http URI (eg a GetFeatureById to the WFS) or you return them in the `wfs:additionalObjects` container and use local references (like `#somecomponentid`).

**Comment by MichalMed** [comment]
There may be a problem, that those components have no geometry elemet, thus they are not 'features' in a way we think about them. So the first option is not applicable  What do you think about it?

**Comment by cportele** [comment]
A feature - in the definition used by OGC/ISO - does not have to have a geometry-valued property. See ISO 19109. The same is true in INSPIRE, see the definitions in the Directive for spatial data and spatial object. Postal descriptors, for example, are clearly "related to a specific location or geographical area", even without a geometry-valued property.

**Comment by MichalMed** [comment]
Maybe I don't get the real point, but how are you going to call those features then using BBOX? 

I get it, that only request returning tohose features is getFeatureById, which is actualy only one I need for referencing from Address feature. Right?

**Comment by cportele** [comment]
In that case, yes, you cannot use a geometric filter on an AddressComponent. 

(Note: If there would be a navigable association role from AddressComponent to Address, say "address", then you could technically use a geometric filter on the property reference `ad:address/*/ad:position/*/ad:geometry`, but in this case it does not make much sense as most components would reference many Address features.)

That is, if there would have been an identified need to request AddressComponents by geometric filters, a geometric property for that component would have been added.

But not all queries (have to) use a geometric filter. For example, you could query the dataset to get all AdminUnitName on municipality level that start with "P" to get a list of them. Whether such a capability is needed/useful is something the publisher has to decide. If the AdminUnitName features are not advertised in the capabilities, then they will not be available via a Query on the AdminUnitFeature type, only using the GetFeatureById query.
